python/semantic_kernel/ai/open_ai/services/azure_chat_completion.py

```diff
@@ -12,33 +12,59 @@
 class AzureChatCompletion(OpenAIChatCompletion):
     _endpoint: str
     _api_version: str
+    _api_type: str
 
     def __init__(
         self,
         deployment_name: str,
-        endpoint: str,
-        api_key: str,
+        endpoint: Optional[str] = None,
+        api_key: Optional[str] = None,
         api_version: str = "2023-03-15-preview",
         logger: Optional[Logger] = None,
+        use_ad_auth=False,
     ) -> None:
+        """
+        Initialize an AzureChatCompletion backend.
+
+        You must provide:
+        - A deployment_name, endpoint, and api_key (plus, optionally: use_ad_auth)
+
+        :param deployment_name: The name of the Azure deployment. This value
+            will correspond to the custom name you chose for your deployment
+            when you deployed a model. This value can be found under
+            Resource Management > Deployments in the Azure portal or, alternatively,
+            under Management > Deployments in Azure OpenAI Studio.
+        :param endpoint: The endpoint of the Azure deployment. This value
+            can be found in the Keys & Endpoint section when examining
+            your resource from the Azure portal.
+        :param api_key: The API key for the Azure deployment. This value can be
+            found in the Keys & Endpoint section when examining your resource in
+            the Azure portal. You can use either KEY1 or KEY2.
+        :param api_version: The API version to use. (Optional)
+            The default value is "2022-12-01".
+        :param logger: The logger instance to use. (Optional)
+        :param use_ad_auth: Whether to use Azure Active Directory authentication.
+            (Optional) The default value is False.
+        """
         if not deployment_name:
             raise ValueError("The deployment name cannot be `None` or empty")
         if not api_key:
             raise ValueError("The Azure API key cannot be `None` or empty`")
         if not endpoint:
             raise ValueError("The Azure endpoint cannot be `None` or empty")
-        if not endpoint.startswith("https://"):
+        if not endpoint.startswith("https://") and not use_ad_auth:
```

Comment:
why allow http:// with AAD? sending AAD tokens over http will expose them to MITM

Reply:
Good catch, I was testing at one point and swapped endpoint url and api key (and wrongly thought that, in AD contexts, the endpoint URL could look funky; just double checked and, even in AD contexts, it is indeed a normal https:// url)

Fixed!